### PR TITLE
[issue 1902] Add CentOS checks to platform tags

### PR DESCRIPTION
**Pull request by shawndwells**
in support of https://github.com/OpenSCAP/scap-security-guide/issues/1902

Refactors OVAL checks. Needs a careful review to make sure I didn't mess up the criteria operators..

### CODE REVIEW

File: shared/oval/sysctl_kernel_exec_shield.xml [hunk 1]
```diff
@@ -10,13 +10,12 @@
...
     <criteria operator="OR">
-      <criteria operator="AND" comment="system is RHEL6">
-        <extend_definition comment="RHEL6 installed" definition_ref="installed_OS_is_rhel6" />
-        <criterion comment="kernel runtime parameter kernel.exec-shield set to 1" test_ref="test_runtime_sysctl_kernel_exec_shield" />
-        <criterion comment="kernel /etc/sysctl.conf parameter kernel.exec-shield set to 1" test_ref="test_static_sysctl_kernel_exec_shield" />
-      </criteria>
-      <criteria operator="AND">
-        <extend_definition comment="32-bit system" definition_ref="system_info_architecture_x86" />
+      <criteria operator="AND" comment="system is RHEL6-based, is x86-32, then eval kernel.exec-shield">
+        <criteria operator="OR" comment"system is RHEL6 or CentOS6">
```
Comment (yuumasato) [inline shared/oval/sysctl_kernel_exec_shield.xml]: There is a small typo here

Comment (yuumasato) [inline shared/oval/sysctl_kernel_exec_shield.xml]: Platform should be changed to `multi_platform_rhel`.

And this same file claims to be gererated but I could not find its template, so I think we can remove that claim.

Comment (shawndwells) [inline shared/oval/sysctl_kernel_exec_shield.xml]: Updated. IIRC this wasn't included in templating because of the 32 vs 64bit oddities.

**Comment by jan-cerny** [comment]
Making this multi_platform_rhel was a great idea, but it doesn't build.

There's a conflict with `RHEL/5/input/oval/sysctl_kernel_nonexec_stack.xml`, which also contains a definition `test_runtime_sysctl_kernel_exec_shield`.

It would be nice if we could drop the RHEL5 definition and use the shared instead.

**Comment by redhatrises** [comment]
@shawndwells why close this PR?

**Comment by shawndwells** [comment]
On 6/13/17 11:41 AM, redhatrises wrote:
> @shawndwells <https://github.com/shawndwells> why close this PR?

free time to tinker with it =/

**Pull request closed without merging**